pennylane/operation.py

```diff
@@ -2478,3 +2478,19 @@ def defines_diagonalizing_gates(obj):
         except DiagGatesUndefinedError:
             return False
         return True
+
+
+@qml.BooleanFn
+def gen_is_hamiltonian(obj):
+    """Returns ``True`` if an operator has a generator defined and it is a Hamiltonian
+    with more than one term."""
+
+    try:
+        o = obj.generator()
+    except (AttributeError, OperatorPropertyUndefined, GeneratorUndefinedError):
+        return False
+
+    if type(o) == qml.Hamiltonian:
+        if len(o.coeffs) > 1:
```

Comment:
The function name doesn't seem to include this part of the logic and it can be confusing (e.g., when reading the code in `pennylane/transforms/tape_expand.py`).
```py
| (has_gen & ~gen_is_hamiltonian),
```
It could be a surprise that multiple terms are considered if we only read the code above.

It would be worth clearing this up by either:
* Describing that multiple terms are considered by updating the function name (e.g., `gen_is_multi_term_hamiltonian`) or
* Separating this function into two: one that checks for a Hamiltonian and the other that checks the number of terms.